transport/grpc/inbound.go

```diff
@@ -81,6 +66,55 @@ func (i *inbound) Start(service transport.ServiceDetail, d transport.Deps) error
 	return nil
 }
 
+/*
+ServiceDescs in GRPC define how clients can interact with a server and usually have well defined schemas to match
+request methods with service functions.  For our purposes we are avoiding this check by dynamically creating the
+serviceDescs from the YARPC registered procedures.  This allows us to see what methods are available and to create an
+individual routing mechanism to each one.  All the routes will be forwarded to the same "Handler" which will do YARPC
+routing to handle methods and encodings.
+*/
+func getServiceDescs(gHandler handler) *[]grpc.ServiceDesc {
+	var serviceDescs []grpc.ServiceDesc
+
+	// Create separate routes for each service & procedure
+	for service, procs := range getServiceProcedures(gHandler.Registry) {
+		serviceDescs = append(serviceDescs, *createServiceDesc(gHandler, service, procs))
+	}
+
+	// If no service procedures are configured insert a default serviceProcedure
+	if len(serviceDescs) == 0 {
+		serviceDescs = append(serviceDescs, *createServiceDesc(gHandler, "yarpc", []string{"yarpc"}))
+	}
+
+	return &serviceDescs
```

Comment:
Why is this returning a pointer to the slice? You're just dereferencing it
above.